### PR TITLE
Feature: Registration > AgencySelection

### PR DESCRIPTION
**Pull request by koepferd**
## Proposed Changes

  - changed & improved user interface for `<AgencySelection />`
  - implemented new interface into `<FormAccordion />`
  - implemented `<FormAccordion />` data handling to submit a registration
  - refactored `<LoadingSpinner />`
  - refactored & expanded `<RadioButton />`
  - removed deprecated styles & logic

### CODE REVIEW

File: src/components/agencySelection/AgencySelection.tsx [hunk 1]
```diff
@@ -14,19 +14,28 @@ import {
...
+	'registration.agencySelection.intro.point2',
+	'registration.agencySelection.intro.point3'
+];
 
 export interface AgencySelectionProps {
 	selectedConsultingType: number | undefined;
 	icon?: JSX.Element;
-	setAgency: Function;
+	onAgencyChange: Function;
+	onValidityChange?: Function;
```
Comment (iam-robin) [inline src/components/agencySelection/AgencySelection.tsx:32]: is there a reason why the callback validation prop is optional?

File: src/components/formAccordion/FormAccordion.tsx [hunk 2]
```diff
@@ -1,21 +1,67 @@
...
+	prefilledAgencyData: any;
+	handleFormAccordionData: Function;
+}
+
+export const FormAccordion = (props: FormAccordionProps) => {
 	const [activeItem, setActiveItem] = useState<number>(1);
 	const [isUsernameValid, setIsUsernameValid] = useState<
 		AccordionItemValidity
 	>('initial');
+	const [username, setUsername] = useState<string>(undefined);
```
Comment (iam-robin) [inline src/components/formAccordion/FormAccordion.tsx]: undefined does not have to be set specifically, it's the default.
Either declare the state with null or leave empty

File: src/components/formAccordion/FormAccordion.tsx [hunk 3]
```diff
@@ -1,21 +1,67 @@
...
+	const [isSelectedAgencyValid, setIsSelectedAgencyValid] = useState<
+		AccordionItemValidity
+	>('initial');
+	const [agency, setAgency] = useState<{ id; postcode }>(undefined);
```
Comment (iam-robin) [inline src/components/formAccordion/FormAccordion.tsx]: same as line 23

File: src/components/agencySelection/AgencySelection.tsx (hunk 1 above)
Comment (koepferd) [reply src/components/agencySelection/AgencySelection.tsx:32]: Yes, the `<AgencySelection />` is not always used in the context of `<FormAccordion />`. `onValidityChange`-Callback is only used for the accordion logic.

**Review by iam-robin — APPROVED** [review]

File: src/components/formAccordion/FormAccordion.tsx [hunk 4]
```diff
@@ -24,17 +62,35 @@ export const FormAccordion = () => {
...
+					onAgencyChange={(agency) => setAgency(agency)}
+					onValidityChange={(validity) =>
+						setIsSelectedAgencyValid(validity)
+					}
+				/>
+			),
+			isValid: isSelectedAgencyValid
+		});
+	}
+	// } else if (autoselectPostcodeForConsultingType(props.consultingType)) {
```
Comment (iam-robin) [inline src/components/formAccordion/FormAccordion.tsx]: comment still needed?

**Review by iam-robin — DISMISSED** [review]

**Review by iam-robin — DISMISSED** [review]

**Review by iam-robin — APPROVED** [review]

**Pull request merged**